pkg/sql/catalog/schemaexpr/partial_index.go

```diff
@@ -59,7 +59,7 @@ func ValidatePartialIndexPredicate(
 	if err != nil {
 		return "", err
 	}
-	if !desc.IsNew() {
+	if mutDesc, ok := desc.(catalog.MutableTableDescriptor); !ok || !mutDesc.IsNew() {
```

Comment:
Perhaps checking that the version > 1 would be equivalent? Just a thought.

Reply:
I don't think it would be equivalent since it doesn't guarantee that the object was created in the current transaction. i.e. It's valid to modify the first version of the object

Reply:
Good point!